### DISCUSSION TITLE
PEP 677 -- Callable Type Syntax.

### DISCUSSION THREAD

**Discussion by smeden-lod** [python/typing#979] [General]
Hi guys. I came across this PEP announcement today, I'm so excited ! I was fighting with the awfull notation for Callable, wondering if there were a chance for getting clean function signatures as the ones in Haskell. And here it is !!! Do you know of any intent to implement that in pyright ?

**Comment by JelleZijlstra** [comment]
Judging from past experience, it's likely that it will be implemented in pyright pretty soon. :)

It might not help that much, though, because type annotation syntax also needs to be valid at runtime (even with `from __future__ import annotations`), and the new syntax will go into Python 3.11 at the earliest. Depending on how pyright implements its parser, you may be able to use the new syntax in quoted annotations though (like `def f(callback: "() -> bool") -> None: ...`).

**Reply by smeden-lod** [reply]
It's still good news though and I admire pyright team ' reactivity !!!

**Reply by smeden-lod** [reply]
I'd like to participate to the dev of the parser, i must admit; I don't konw how, but...

**Reply by erictraut** [reply]
I was just waiting for it to reach the point where it was accepted as a draft PEP and given a real PEP number. Looks like that has now happened. I just filed an enhancement request for myself: https://github.com/microsoft/pyright/issues/2720.

As @JelleZijlstra said, you won't be able to use this in runtime executable statements until 3.11 at the earliest, but it will be usable in type stubs and quoted type annotations in the meantime.

**Reply by erictraut** [reply]
I just published pyright 1.1.197, which includes support for the new PEP 677 syntax. I'd appreciate bug reports for any issues you find.

**Reply by smeden-lod** [reply]
Great ! Took on testing the feature on the examples given in the PEP. Works fine the first one (`flatmap`, `wrap` and `add`
